src/server/terrain/structure_building_blocks.zig

```diff
@@ -1,25 +1,31 @@
 const std = @import("std");
 
-const main = @import("root");
+// const main = @import("root");
+const main = @import("../../main.zig");
 const ZonElement = main.ZonElement;
 const Blueprint = main.blueprint.Blueprint;
 const List = main.List;
 const ListUnmanaged = main.ListUnmanaged;
 const AliasTable = main.utils.AliasTable;
 const Neighbor = main.chunk.Neighbor;
 const Block = main.blocks.Block;
-const parseBlock = main.blocks.parseBlock;
 const Degrees = main.rotation.Degrees;
-const hashInt = main.utils.hashInt;
 const NeverFailingAllocator = main.heap.NeverFailingAllocator;
 
 var arena = main.heap.NeverFailingArenaAllocator.init(main.globalAllocator);
 const arenaAllocator = arena.allocator();
 
 var structureCache: std.StringHashMapUnmanaged(StructureBuildingBlock) = .{};
-var blueprintCache: std.StringHashMapUnmanaged([4]*BlueprintEntry) = .{};
+var blueprintCache: std.StringHashMapUnmanaged([4]BlueprintEntry) = .{};
```

Comment:
Should this be `*[4]BlueprintEntry`? Otherwise we end up copying it in every place it's used, and BlueprintEntry is not a small struct.